src/features/projectsV2/ProjectSnippet/index.tsx

```diff
@@ -114,22 +112,63 @@ export default function ProjectSnippet({
     country: project.country,
     currency: project.currency,
   };
+  return (
+    <>
+      <ImageSection {...imageProps} />
+      <div className={styles.progressBar}>
+        <div
+          className={`${styles.progressBarHighlight} ${progressBarClass}`}
+          style={{ width: `${progressPercentage}%` }}
+        />
+      </div>
+      <ProjectInfoSection {...projectInfoProps} />
+    </>
+  );
+};
+
+export default function ProjectSnippet({
+  project,
+  showTooltipPopups,
+  isMobile,
+  page,
+}: Props): ReactElement {
+  const { embed, callbackUrl } = useContext(ParamsContext);
+  const locale = useLocale();
+  const isTopProject = project.purpose === 'trees' && project.isTopProject;
+  const isApproved = project.purpose === 'trees' && project.isApproved;
+
   const projectSnippetContainerClasses = `${styles.singleProject} ${
     page === 'project-details' && isMobile
       ? styles.projectDetailsSnippetMobile
       : ''
   }`;
+  const ProjectSnippetContentProps = {
+    showTooltipPopups,
+    page,
+    project,
+  };
+  const getProjectPath = useCallback(() => {
+    let path = `/${locale}/prd/${project.slug}`;
+    if (embed === 'true') {
+      const params = new URLSearchParams({ embed: 'true' });
+      if (callbackUrl !== undefined && typeof callbackUrl === 'string') {
+        params.append('callback', callbackUrl);
+      }
+      path += `?${params.toString()}`;
+    }
+    return path;
+  }, [page, locale, project.slug, embed, callbackUrl]);
   return (
     <>
       <div className={projectSnippetContainerClasses}>
-        <ImageSection {...imageProps} />
-        <div className={styles.progressBar}>
-          <div
-            className={`${styles.progressBarHighlight} ${progressBarClass}`}
-            style={{ width: `${progressPercentage}%` }}
-          />
-        </div>
-        <ProjectInfoSection {...projectInfoProps} />
+        {page === 'project-details' ? (
+          <ProjectSnippetContent {...ProjectSnippetContentProps} />
+        ) : (
+          <Link href={getProjectPath()} style={{ cursor: 'pointer' }}>
+            <ProjectSnippetContent {...ProjectSnippetContentProps} />
+          </Link>
```

Comment:
Avoid calling functions from JSX, unless the function is an event handler. It is harder to debug, and could cause unexpected behavior. Here, you could store `projectPath` in a memoized variable and directly pass it to `href`